node/node.go

```diff
@@ -251,6 +275,39 @@ func (n *NodeController) Ready() <-chan struct{} {
 	return n.chReady
 }
 
+// pingLoop does one thing. It pings the provider. It does this until context is cancelled.
+func (n *NodeController) pingLoop(ctx context.Context) {
+	realCheckFunc := func(ctx context.Context) {
+		ctx, span := trace.StartSpan(ctx, "node.pingNode")
+		defer span.End()
+		err := n.p.Ping(ctx)
+		n.lastPingErrorLock.Lock()
+		n.lastPingError = err
+		n.lastPingErrorLock.Unlock()
+		span.SetStatus(err)
+	}
+
+	// We should have a context that we cancel when we pass to the node, this wrapper just determines whether
+	// or not it will be one with a timeout
+	wrappedCheckFunc := func(ctx context.Context) {
+		if n.pingTimeout != nil {
+			ctx, cancel := context.WithTimeout(ctx, *n.pingTimeout)
```

Comment:
I'm not sure about the timeout actually.
Because there is nothing we can do in this loop to ensure that the ping func actually cancels on timeout.
Meanwhile the node provider can define its own timeout.

Reply:
We can leave it out for now.